Feat: get display name from SAML

dorianim opened the pull request:
This PR adds the capability of getting a users display name from a SAML provider.
What do you think?

campbell-m commented:
Thanks for this.  My comments are:

1. I think systemdefaults.inc.php also needs to be updated to include the two extra `$auth` attributes.
2. Is line 130 (` // Returns an empty string if no e-mail address was found`) a copy and paste typo?  Should it say 'displayname' instead of 'email-address'?
3. `      return $givenName . ' ' . $surname;` should probably be something like `      return trim($givenName . ' ' . $surname);` to cater for the cases where `$givenName` and/or `$surname` are the empty string.
4. The code assumes that displaynames are represented in the form "[givenName] [surname]".  However in some locales they are represented as "[surname] [givenName]" (or possibly even "[surname], [givenName]").  I'm not sure at the moment what the best way of handling this should be.  Maybe it can wait for now and be addressed in a future PR.

dorianim commented:
Fixed 1,2 and 3. I think, 4 should be handled in a different PR once there is somebody who actually needs it.

Btw.: I just added a builtin simplesamlphp to my dockerimage which makes it super easy to configure using just some variables in the config.inc.php :)
https://github.com/dorianim/mrbs-docker/tree/main#saml-authentication
